sentinelhub/api/batch/utils.py

```diff
@@ -50,11 +112,77 @@ def monitor_batch_job(
     """
     if sleep_time < _MIN_SLEEP_TIME:
         raise ValueError(f"To avoid making too many service requests please set sleep_time>={_MIN_SLEEP_TIME}")
+    batch_kind = _auto_adjust_kind(batch_request, batch_kind)
 
-    batch_request = monitor_batch_analysis(batch_request, config, sleep_time=analysis_sleep_time)
+    batch_request = monitor_batch_analysis(batch_request, config, sleep_time=analysis_sleep_time, batch_kind=batch_kind)
     if batch_request.status is BatchRequestStatus.PROCESSING:
         LOGGER.info("Batch job is running")
 
+    if isinstance(batch_request, BatchRequest):
+        return _monitor_batch_process_execution(batch_request, config, sleep_time)
+    return _monitor_batch_statistical_execution(batch_request, config, sleep_time)  # type: ignore[func-returns-value]
+
+
+@overload
+def monitor_batch_analysis(
+    batch_request: BatchRequestSpec,
+    config: Optional[SHConfig],
+    sleep_time: int = _DEFAULT_ANALYSIS_SLEEP_TIME,
+    *,
+    batch_kind: Literal["process"],
+) -> BatchRequest:
+    pass
+
+
+@overload
+def monitor_batch_analysis(
+    batch_request: BatchRequestSpec,
+    config: Optional[SHConfig],
+    sleep_time: int = _DEFAULT_ANALYSIS_SLEEP_TIME,
+    *,
+    batch_kind: Literal["statistical"],
+) -> BatchStatisticalRequest:
+    pass
+
+
+def monitor_batch_analysis(
+    batch_request: BatchRequestSpec,
+    config: Optional[SHConfig] = None,
+    sleep_time: int = _DEFAULT_ANALYSIS_SLEEP_TIME,
+    *,
+    batch_kind: BatchKind = "process",
```

Comment:
I would actually put a default `batch_kind: BatchKind = "auto"`. In this case, the kind would be determined from `batch_request` (if class is given) or an error would be raised.